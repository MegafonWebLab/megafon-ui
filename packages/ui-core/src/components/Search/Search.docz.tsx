import * as React from 'react';

interface IDemoSearchWrapperProps {
    children: (prop: {
        onSubmit: (value: string) => void;
        onChange: (query: string) => void;
        chosenValue: string;
        items: string[];
    }) => JSX.Element;
}

export const DemoSearchWrapper: React.FC<IDemoSearchWrapperProps> = ({ children }) => {
    const [value, setValue] = React.useState('');
    const [items, setItems] = React.useState<string[]>([]);

    const handleChange = React.useCallback((query: string) => {
        if (!query) {
            return;
        }

        setItems((oldItems) => [...oldItems, query]);
    }, [items, setItems]);

    return (
        <div>
            <p style={{ marginTop: 0 }}>Value: {value}</p>
            {children({ onChange: handleChange, onSubmit: setValue, items: items, chosenValue: value })}
        </div>
    );
};
