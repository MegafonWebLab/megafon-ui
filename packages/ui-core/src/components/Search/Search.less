@import '~styles/base.less';
@b: e('.mfui-beta-search');

@{b} {
    .fontFamily();
    .commonFont();
    position: relative;
    z-index: auto;
    color: @freshAsphalt;
    min-height: 40px;

    &__control {
        display: flex;
        width: 100%;
        box-sizing: border-box;
        position: relative;
        z-index: 1;
        height: 48px;
        cursor: pointer;
        border: 1px solid @spbSky2;
        background-color: @clearWhite;
        outline: none;
        padding: 0 20px;
    }

    &__search-field {
        box-sizing: border-box;
        font-family: inherit;
        line-height: 25px;
        position: relative;
        z-index: 1;
        background-color: transparent;
        border: none;
        height: 100%;
        //fix for ios - Focus on input with padding (filled with text) pushes cursor out of bounds
        padding: 0;
        outline: none;
        opacity: 1;
        border-radius: 0;
        -webkit-appearance: none;
        -moz-appearance: none;
        box-shadow: none;
        overflow: auto;
        flex-grow: 1;

        @media @mobileSM {
            width: 90%;
        }
    }

    &__list {
        display: none;
        position: absolute;
        top: 100%;
        left: 0;
        right: 0;
        z-index: 0;
        background-color: @clearWhite;
        box-shadow: 0 1px 16px rgba(0, 0, 0, 0.2);
    }

    &__icon {
        min-width: 24px;
        width: 24px;
        fill: @freshAsphalt;
    }

    &__icon-box {
        display: flex;
        justify-content: center;
        height: 100%;
    }

    &__list-inner {
        overflow: auto;
        padding: 12px 0;
        box-sizing: border-box;
    }

    &__list-item {
        .fontFamily();
        padding: 8px 32px;
        cursor: pointer;
        background-color: @clearWhite;
    }

    &__list-item_active {
        background-color: @spbSky0;
    }

    &__item-title {
        overflow: hidden;
        white-space: nowrap;
        flex-grow: 1;
        position: relative;

        &:after {
            content: '';
            position: absolute;
            top: 0;
            right: 0;
            bottom: 0;
            width: 40px;
            background: linear-gradient(to right, rgba(255, 255, 255, 0), @clearWhite);
        }
    }

    &__list-item_active &__item-title {
        &:after {
            background: linear-gradient(to right, rgba(255, 255, 255, 0), @spbSky0);
        }
    }

    &_open &__list {
        display: block;
        border-color: @freshAsphalt;
    }

    &_open {
        z-index: 12;
    }

    &__highlighted-fragment {
        font-weight: bold;
    }
}
