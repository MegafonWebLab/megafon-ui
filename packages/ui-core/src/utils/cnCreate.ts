import classnames from 'classnames';

interface IModificators {
    [key: string]: boolean | string | undefined | null;
}

interface IClassSet {
    [key: string]: boolean;
}

type CustomClassNameType = string;
type CustomClassNamesArrayType = string[];
type CustomClassNamesType = CustomClassNameType | CustomClassNamesArrayType | undefined;

type ElementNameType = string | IModificators | CustomClassNamesArrayType | CustomClassNamesType;

const getCustomClassName = (first?: ElementNameType, second?: ElementNameType, third?: CustomClassNamesType) => {
    if (Array.isArray(first)) {
        return first.filter(f => typeof f === 'string').join(' ');
    }
    if (Array.isArray(second)) {
        return second.filter(f => typeof f === 'string').join(' ');
    }
    if (typeof second === 'string') {
        return second;
    }
    if (Array.isArray(third)) {
        return third.filter(f => typeof f === 'string').join(' ');
    }
    if (typeof third === 'string') {
        return third;
    }

    return '';
};

/**
 * Бэм генератор
 * @param {String} blockName - название блока
 * @returns {function} - возвращает функцию с тремя возожными параметрами
 * @param {String} elementName - название элемента
 * @param {Object} modificatorsObject - объект с модификаторами
 * @param {String} customClassNames - названия кастомных классов
 * @returns {String}
 */
function cnCreate(blockName: string) {
    function getElement(): string;
    function getElement(elementName: ElementNameType): string;
    function getElement(modificatorsObject: ElementNameType): string;
    function getElement(customClassNames: CustomClassNamesArrayType): string;
    function getElement(elementName: ElementNameType, modificatorsObject: ElementNameType): string;
    function getElement(elementName: ElementNameType, customClassNames: CustomClassNamesType): string;
    function getElement(modificatorsObject: ElementNameType, customClassNames: CustomClassNamesType): string;
    function getElement(
        elementName: ElementNameType,
        modificatorsObject: ElementNameType,
        customClassNames: CustomClassNamesType
    ): string;
    function getElement(
        elementName?: ElementNameType,
        modificatorsObject?: ElementNameType,
        customClassNames?: CustomClassNamesType
    ): string {
        const prefix = typeof elementName === 'string' ? elementName : '';
        const className = getCustomClassName(elementName, modificatorsObject, customClassNames);
        const secondMods = !Array.isArray(modificatorsObject) && typeof modificatorsObject !== 'string'
            ? modificatorsObject
            : undefined;
        const params = typeof elementName !== 'string' && !Array.isArray(elementName)
            ? elementName
            : secondMods;

        if (!params || typeof params === 'object' && Object.keys(params).length === 0) {
            return classnames(`${blockName}${prefix ? '__' + prefix : ''}`, className);
        }

        const classParams: IClassSet = {};
        let prefixKey = '', withoutPrefix = '';

        for (const key in params) {
            if (typeof params[key] === 'boolean' && params[key]) {
                prefixKey = `__${prefix}_${key}`;
                classParams[
                    `${blockName}${prefix ? prefixKey : '_' + key}`
                ] = true;
            } else if (typeof params[key] === 'string') {
                prefixKey = `__${prefix}_${key}_${params[key]}`;
                withoutPrefix = `_${key}_${params[key]}`;
                classParams[
                    `${blockName}${prefix ? prefixKey : withoutPrefix}`
                ] = true;
            }
        }

        return prefix === ''
            ? classnames(blockName, classParams, className)
            : classnames(`${blockName}__${prefix}`, classParams, className);
    }

    return getElement;
}

export default cnCreate;
