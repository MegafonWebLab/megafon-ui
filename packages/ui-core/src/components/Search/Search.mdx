---
name: Search
menu: Компоненты
route: components/search
---

import { Playground, Props } from 'docz';
import Search from './Search';
import { DemoSearchWrapper } from './Search.docz';

# Search

## Properties

<Props of={Search} />

```javascript
export const DemoSearchWrapper: React.FC<IDemoSearchWrapperProps> = ({ children }) => {
    const [value, setValue] = React.useState('');
    const [items, setItems] = React.useState([]);

    const handleChange = React.useCallback((query: string) => {
        if (!query) {
            return;
        }

        setItems((oldItems) => [...oldItems, query]);
    }, [items, setItems]);

    return (
        <div>
            <p style={{ marginTop: 0 }}>Value: {value}</p>
            {children({ onChange: handleChange, onSubmit: setValue, items: items, chosenValue: value })}
        </div>
    );
};
```

## Basic usage

<Playground>
    <DemoSearchWrapper>
        {({ onSubmit }) =>
            <Search
                onSubmit={onSubmit}
                placeholder="Type here and press ENTER"
            />
        }
    </DemoSearchWrapper>
</Playground>

## Usage without icon

<Playground>
    <Search hideIcon />
</Playground>

## Usage with initial value

<Playground>
    <DemoSearchWrapper>
        {({ onSubmit }) =>
            <Search
                placeholder="Type here for search"
                value="Previous request"
                onSubmit={onSubmit}
            />
        }
    </DemoSearchWrapper>
</Playground>

## Usage with onChange callback

<Playground>
    <DemoSearchWrapper>
        {({ onSubmit }) =>
            <Search
                placeholder="Type here for search"
                onChange={onSubmit}
                changeDelay={50}
            />
        }
    </DemoSearchWrapper>
</Playground>

## Usage with emulating incomingData

<Playground>
    <DemoSearchWrapper>
        {({ onSubmit, chosenValue, items, onChange }) => (
            <Search
                value={chosenValue}
                items={items}
                placeholder="Type and wait for results"
                onChange={onChange}
                onSubmit={onSubmit}
            />
        )}
    </DemoSearchWrapper>
</Playground>
