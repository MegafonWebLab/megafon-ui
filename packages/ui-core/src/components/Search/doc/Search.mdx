---
name: Search
menu: Компоненты
route: components/search
---

import { Props, Playground } from 'docz';
import Search from '../Search';
import Example from './Search.example.mdx';

# Search

<CopyToClipboardBox text="import { Search } from '@megafon/ui-core';" />

<DoczTabs>
    <Example />
    <Props of={Search} />
</DoczTabs>
