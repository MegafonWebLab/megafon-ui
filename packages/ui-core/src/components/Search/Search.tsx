import * as React from 'react';
import * as PropTypes from 'prop-types';
import { useCallback, useState, useRef, useEffect } from 'react';
import SearchIcon from 'icons/Basic/16/Search_16.svg';
import debounce from 'lodash.debounce';
import cnCreate from 'utils/cnCreate';
import './Search.less';

type HandleSearchSubmit = (e?: React.MouseEvent<HTMLDivElement>) => void;
type HandleSelectSubmit = (i: number) => (e: React.MouseEvent) => void;
type HandleItemSubmit = (index: number) => void;

export interface ISearchProps {
    /** Current value */
    value?: string;
    /** Placeholder */
    placeholder?: string;
    /** Forcefully prohibits icon's render */
    hideIcon?: boolean;
    /** Array of strings to be used for options rendering */
    items?: string[];
    /** Debounce delay for onChange callback */
    changeDelay?: number;
    /** Change handler */
    onChange?: (value: string) => void;
    /** Submit handler */
    onSubmit?: (value: string) => void;
     /** Custom className */
    className?: string;
}

const cn = cnCreate('mfui-beta-search');
const Search: React.FC<ISearchProps> = ({
        value = '',
        placeholder,
        hideIcon,
        items = [],
        changeDelay = 250,
        className,
        onChange,
        onSubmit,
}) => {
    const [searchQuery, setSearchQuery] = useState(value);
    const [activeIndex, setActiveIndex] = useState(-1);
    const [isFocused, setFocus] = useState(false);
    const debouncedOnChange = useRef(debounce((inputValue) => onChange && onChange(inputValue), changeDelay));

    useEffect(() => setSearchQuery(value), [value, setSearchQuery]);

    const handleChange: React.EventHandler<React.ChangeEvent<HTMLInputElement>> = (e) => {
        const { target: { value: inputValue = '' } } = e;

        setSearchQuery(inputValue);
        setActiveIndex(-1);

        changeDelay === 0 ? onChange && onChange(inputValue) : debouncedOnChange.current(inputValue);
    };

    const handleHoverItem = useCallback((index: number) => (_e: React.SyntheticEvent<EventTarget>): void => {
        setActiveIndex(index);
    }, [setActiveIndex]);

    const handleSearchSubmit: HandleSearchSubmit = useCallback((): void => {
        onSubmit && searchQuery && onSubmit(searchQuery);
    }, [searchQuery, onSubmit]);

    const handleItemSubmit: HandleItemSubmit = useCallback((index: number): void => {
        const chosenValue = items[index];

        onSubmit && onSubmit(chosenValue);
    }, [onSubmit, items]);

    const handleSelectSubmit: HandleSelectSubmit =
        useCallback(() => (_e): void => {
            handleItemSubmit(activeIndex);
        }, [handleItemSubmit, activeIndex]);

    const handleFieldFocusToggle = useCallback((): void => {
        setFocus(focus => !focus);
    }, [setFocus]);

    const handleClick = useCallback((): void => {
        if (activeIndex >= 0) {
            setActiveIndex(-1);
        }
    }, [activeIndex, setActiveIndex]);

    const handleKeyDown = useCallback((e: React.KeyboardEvent<HTMLDivElement>): boolean => {
        if (e.key === 'ArrowDown' && activeIndex < items.length - 1) {
            setActiveIndex(index => index + 1);
            e.preventDefault();
        } else if (e.key === 'ArrowUp' && activeIndex > -1) {
            setActiveIndex(index => index - 1);
            e.preventDefault();
        } else if (e.key === 'Enter' && activeIndex > -1) {
            handleItemSubmit(activeIndex);
        } else if (e.key === 'Enter' && activeIndex === -1) {
            handleSearchSubmit();
        }

        return false;
    }, [activeIndex, setActiveIndex, handleSearchSubmit, handleItemSubmit]);

    const highlightString = (title) => {
        const stringFragments = title.split(RegExp(`(${searchQuery})`, 'ig'));

        return (
            <>
                {stringFragments.map((fragment, i) => (
                    <React.Fragment key={i}>
                        {(fragment.toLowerCase() === searchQuery.toLowerCase())
                            ? <span className={cn('highlighted-fragment')}>{fragment}</span>
                            : fragment
                        }
                    </React.Fragment>
                ))}
            </>
        );
    };

    return (
        <div className={cn({ open: isFocused }, [className])}>
            <div className={cn('control')}>
                <input
                    className={cn('search-field')}
                    placeholder={placeholder}
                    value={searchQuery}
                    onChange={handleChange}
                    onFocus={handleFieldFocusToggle}
                    onBlur={handleFieldFocusToggle}
                    onKeyDown={handleKeyDown}
                    onClick={handleClick}
                    type="text"
                    autoComplete="off"
                />
                {!hideIcon && <div
                    className={cn('icon-box')}
                    onClick={handleSearchSubmit}
                >
                    <SearchIcon className={cn('icon')} />
                </div>}
            </div>
            {items && !!items.length &&
                <div className={cn('list')}>
                    <div className={cn('list-inner')}>
                        {items.map((title, i) =>
                            <div
                                className={cn('list-item', { active: activeIndex === i })}
                                onMouseDown={handleSelectSubmit(i)}
                                onMouseEnter={handleHoverItem(i)}
                                key={i}
                            >
                                <div className={cn('item-title')}>
                                    {highlightString(title)}
                                </div>
                            </div>
                        )}
                    </div>
                </div>
            }
        </div>
    );
};

Search.propTypes = {
    value: PropTypes.string,
    placeholder: PropTypes.string,
    hideIcon: PropTypes.bool,
    changeDelay: PropTypes.number,
    items: PropTypes.arrayOf(PropTypes.string.isRequired),
    onChange: PropTypes.func,
    onSubmit: PropTypes.func,
};

export default Search;
