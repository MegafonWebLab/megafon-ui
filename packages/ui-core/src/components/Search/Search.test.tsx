import * as React from 'react';
import { shallow, mount } from 'enzyme';
import Search , { ISearchProps } from './Search';
import cnCreate from 'utils/cnCreate';

const cn = cnCreate('mfui-beta-search');

const props: ISearchProps = {
    value: 'initial value',
    placeholder: 'type to search here',
    items: ['title', 'title2', 'title3', 'title4'],
    className: 'test',
};

describe('<Search />', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    describe('snapshots', () => {
        it('renders Search without props correctly', () => {
            const wrapper = shallow(<Search />);
            expect(wrapper).toMatchSnapshot();
        });

        it('renders Search with all props correctly', () => {
            const wrapper = shallow(<Search {...props} />);
            expect(wrapper).toMatchSnapshot();
        });

        it('renders Search without icon', () => {
            const wrapper = shallow(<Search {...props} hideIcon />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('typing', () => {
        it('highlighted text', () => {
            const wrapper = shallow(<Search {...props} />);

            wrapper.find(`.${cn('search-field')}`).simulate('change', {
                target: {
                    value: 'title2',
                },
            });

            expect(wrapper).toMatchSnapshot();
        });
        it('calls onChange while typing in the field with delay', async () => {
            const handleChange = jest.fn();
            const wrapper = mount(<Search {...props} changeDelay={300} onChange={handleChange} />);

            wrapper.find(`.${cn('search-field')}`).simulate('change', {
                target: {
                    value: 'new value',
                },
            });

            expect(handleChange).not.toHaveBeenCalled();

            await new Promise((r) => setTimeout(r, 150));

            expect(handleChange).not.toHaveBeenCalled();

            await new Promise((r) => setTimeout(r, 150));

            expect(handleChange).toHaveBeenCalledWith('new value');
        });
        it('calls onChange', () => {
            const handleChange = jest.fn();
            const wrapper = shallow(<Search {...props} changeDelay={0} onChange={handleChange} />);

            wrapper.find(`.${cn('search-field')}`).simulate('change', {
                target: {
                    value: 'new value',
                },
            });

            expect(handleChange).toBeCalledWith('new value');
        });
    });

    describe('submitting', () => {
        it('calls onSubmit on icon click', () => {
            const handleSubmit = jest.fn();
            const wrapper = shallow(<Search {...props} onSubmit={handleSubmit} />);

            wrapper.find(`.${cn('icon-box')}`).simulate('click');

            expect(handleSubmit).toBeCalledWith(props.value);
        });

        it('calls onSubmit on press Enter with focus on input', () => {
            const handleSubmit = jest.fn();
            const wrapper = shallow(<Search {...props} onSubmit={handleSubmit} />);

            wrapper.find(`.${cn('search-field')}`).simulate('keyDown', { key: 'Enter' });

            expect(handleSubmit).toBeCalledWith(props.value);
        });

        it('calls onSubmit on press Enter with value from items', () => {
            const handleSubmit = jest.fn();
            const wrapper = mount(<Search {...props} onSubmit={handleSubmit} />);

            wrapper.find(`.${cn('search-field')}`).simulate('change', {
                target: {
                    value: 'title',
                },
            });

            wrapper.find(`.${cn('list-item')}`).at(0).simulate('mouseenter');
            wrapper.find(`.${cn('search-field')}`).simulate('keyDown', { key: 'Enter' });

            expect(handleSubmit).toBeCalledWith('title');
        });

        it('calls onSubmit by clicking on item', () => {
            const handleSubmit = jest.fn();
            const listItem = `.${cn('list-item')}`;
            const wrapper = mount(<Search {...props} onSubmit={handleSubmit} />);

            wrapper.find(`.${cn('search-field')}`).simulate('change', {
                target: {
                    value: 'title',
                },
            });
            wrapper.find(listItem).at(0).simulate('mouseenter');
            wrapper.find(listItem).at(0).simulate('mousedown');

            expect(handleSubmit).toBeCalledWith('title');
        });
    });

    describe('keyboard actions', () => {
        it('handles ArrowUp and ArrowDown correctly', () => {
            const wrapper = mount(<Search {...props} />);
            const searchField = wrapper.find(`.${cn('search-field')}`);
            const listItem = `.${cn('list-item')}`;
            const listItemActive = `${cn('list-item_active')}`;

            searchField.simulate('change', {
                target: {
                    value: 'title',
                },
            });

            searchField.simulate('keyDown', { key: 'ArrowDown', preventDefault: () => {}});
            expect(wrapper.find(listItem).at(0).hasClass(listItemActive)).toEqual(true);

            searchField.simulate('keyDown', { key: 'ArrowDown', preventDefault: () => {}});
            expect(wrapper.find(listItem).at(0).hasClass(listItemActive)).toEqual(false);
            expect(wrapper.find(listItem).at(1).hasClass(listItemActive)).toEqual(true);

            searchField.simulate('keyDown', { key: 'ArrowUp', preventDefault: () => {}});
            expect(wrapper.find(listItem).at(0).hasClass(listItemActive)).toEqual(true);
            expect(wrapper.find(listItem).at(1).hasClass(listItemActive)).toEqual(false);
        });
    });
});
