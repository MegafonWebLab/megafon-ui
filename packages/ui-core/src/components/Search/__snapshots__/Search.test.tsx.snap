// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> snapshots renders Search with all props correctly 1`] = `
<div
  className="mfui-beta-search test"
>
  <div
    className="mfui-beta-search__control"
  >
    <input
      autoComplete="off"
      className="mfui-beta-search__search-field"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="type to search here"
      type="text"
      value="initial value"
    />
    <div
      className="mfui-beta-search__icon-box"
      onClick={[Function]}
    >
      <test-file-stub
        className="mfui-beta-search__icon"
      />
    </div>
  </div>
  <div
    className="mfui-beta-search__list"
  >
    <div
      className="mfui-beta-search__list-inner"
    >
      <div
        className="mfui-beta-search__list-item"
        key="0"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="mfui-beta-search__item-title"
        >
          title
        </div>
      </div>
      <div
        className="mfui-beta-search__list-item"
        key="1"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="mfui-beta-search__item-title"
        >
          title2
        </div>
      </div>
      <div
        className="mfui-beta-search__list-item"
        key="2"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="mfui-beta-search__item-title"
        >
          title3
        </div>
      </div>
      <div
        className="mfui-beta-search__list-item"
        key="3"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="mfui-beta-search__item-title"
        >
          title4
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Search /> snapshots renders Search without icon 1`] = `
<div
  className="mfui-beta-search test"
>
  <div
    className="mfui-beta-search__control"
  >
    <input
      autoComplete="off"
      className="mfui-beta-search__search-field"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="type to search here"
      type="text"
      value="initial value"
    />
  </div>
  <div
    className="mfui-beta-search__list"
  >
    <div
      className="mfui-beta-search__list-inner"
    >
      <div
        className="mfui-beta-search__list-item"
        key="0"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="mfui-beta-search__item-title"
        >
          title
        </div>
      </div>
      <div
        className="mfui-beta-search__list-item"
        key="1"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="mfui-beta-search__item-title"
        >
          title2
        </div>
      </div>
      <div
        className="mfui-beta-search__list-item"
        key="2"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="mfui-beta-search__item-title"
        >
          title3
        </div>
      </div>
      <div
        className="mfui-beta-search__list-item"
        key="3"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="mfui-beta-search__item-title"
        >
          title4
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Search /> snapshots renders Search without props correctly 1`] = `
<div
  className="mfui-beta-search"
>
  <div
    className="mfui-beta-search__control"
  >
    <input
      autoComplete="off"
      className="mfui-beta-search__search-field"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      type="text"
      value=""
    />
    <div
      className="mfui-beta-search__icon-box"
      onClick={[Function]}
    >
      <test-file-stub
        className="mfui-beta-search__icon"
      />
    </div>
  </div>
</div>
`;

exports[`<Search /> typing highlighted text 1`] = `
<div
  className="mfui-beta-search test"
>
  <div
    className="mfui-beta-search__control"
  >
    <input
      autoComplete="off"
      className="mfui-beta-search__search-field"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="type to search here"
      type="text"
      value="title2"
    />
    <div
      className="mfui-beta-search__icon-box"
      onClick={[Function]}
    >
      <test-file-stub
        className="mfui-beta-search__icon"
      />
    </div>
  </div>
  <div
    className="mfui-beta-search__list"
  >
    <div
      className="mfui-beta-search__list-inner"
    >
      <div
        className="mfui-beta-search__list-item"
        key="0"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="mfui-beta-search__item-title"
        >
          title
        </div>
      </div>
      <div
        className="mfui-beta-search__list-item"
        key="1"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="mfui-beta-search__item-title"
        >
          <span
            className="mfui-beta-search__highlighted-fragment"
          >
            title2
          </span>
        </div>
      </div>
      <div
        className="mfui-beta-search__list-item"
        key="2"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="mfui-beta-search__item-title"
        >
          title3
        </div>
      </div>
      <div
        className="mfui-beta-search__list-item"
        key="3"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="mfui-beta-search__item-title"
        >
          title4
        </div>
      </div>
    </div>
  </div>
</div>
`;
